command/daemon.go

```diff
@@ -247,9 +253,12 @@ func daemonAction(cctx *cli.Context) error {
 			cfg.Ingest.ResendDirectAnnounce = false
 		}
 
+		var ingestOpts []ingest.Option
+		if cfg.Ingest.IndexCountsEnabled {
+			ingestOpts = append(ingestOpts, ingest.WithIndexCounts(indexCounts))
+		}
```

Comment:
This is not needed, because passing in a nil `indexCounts` is the same as not passing the option at all:
https://github.com/ipni/storetheindex/blob/main/internal/ingest/option.go#L31

Reply:
This if condition is needed because index counter is instantiated if finder flag **or** ingest flag is set.

Reply:
What I am saying is that since either flag instantiates the index counter, always pass `WithIndexCounter(indexCounts)`, and the fact that indexCounts is `nil` or not handles making it optional.